src/polytopes/simplex.jl

```diff
@@ -0,0 +1,41 @@
+"""
+    Simplex(p1, p2, ..., pK, pK+1)
+
+A K-simplex is a polytope with parametric dimension K and K+1 vertices.
+It generalizes the [`Segment`](@ref), the  [`Triangle`](@ref) and the
+[`Tetrahedron`](@ref) to higher dimensions (K > 3).
+
+See also [`issimplex`](@ref).
+"""
+struct Simplex{K,Dim,T,N} <: Polytope{K,Dim,T}
+  vertices::NTuple{N,Point{Dim,T}}
+end
+
+function Simplex(vertices::NTuple{N,Point{Dim,T}}) where {Dim,T,N}
+  K = N - 1
+  K ≤ Dim || throw(ArgumentError("Simplex rank (number of vertices - 1) must be less or equal to embedding dimension."))
+  Simplex{K,Dim,T,N}(vertices)
+end
+
+Simplex(vertices::Point{Dim,T}...) where {Dim,T} = Simplex(vertices)
+Simplex(vertices::Tuple...) = Simplex(Point.(vertices))
+
+# Parametric constructor required for [`materialize`](@ref).
+function Simplex{K}(vertices::NTuple{N}) where {K,N}
+  N == K + 1 || throw(ArgumentError("Number of vertices must be rank K plus one."))
+  Simplex(vertices...)
+end
+function Simplex{K}(vertices::Vararg{Point,N}) where {K,N}
+  N == K + 1 || throw(ArgumentError("Number of vertices must be rank K plus one."))
+  Simplex(vertices...)
+end
```

Comment:
@eliascarv can you please double check that these constructors are following our conventions? I.e., they are defined in terms of the appropriate methods, avoiding unnecessary splatting, and parametric constructors coming first.